python/segyio/segy.py

```diff
@@ -833,6 +833,83 @@ def readonly(self):
 
         return '+' not in self._mode
 
+
+    def interpret(self, ilines, xlines, offsets=None, sorting=2):
+
+        """ (Re)interpret structure on top of a file
+
+        (Re)interprets the structure of the file based in the user supplied
+        iline, xline and offset-indecies. It is required that the number of
+        xlines, ilines and offsets correspond to the tracecount of the file.
+        Note that the actual file itself is not changed, it is only segyio's
+        interpretation of the file that changes.
+
+        Parameters
+        ----------
+
+        f : SegyFile
+        ilines : array_like
+            ilines indecies in new structure
+        xlines : array_like
+            xlines indecies in new structure
+        offset : array_like
+            offset indecies in new structure
+        sorting : int
+            Sorting in new structure
+
+        Notes
+        -----
+
+        .. versionadded:: 1.8
+
+        Examples
+        --------
+        (Re)interpret the structure of the file:
+        >>> ilines = [10, 11, 12, 13]
+        >>> xlines = [20, 21, 22, 23, 24]
+        >>> with segyio.open(file, ignore_geometry=True) as f:
+        ... f.interpret(ilines, xlines)
+        """
+
+        if offsets is None:
+            offsets = np.arange(1)
+
+        ilines = np.asarray(ilines)
+        xlines = np.asarray(xlines)
+        offsets = np.asarray(offsets)
+
+        if np.size(ilines) * np.size(xlines) * np.size(offsets) != self.tracecount:
+            error = "The length of ilines * xlines * offsets does not match the number of traces"
+            raise ValueError(error)
```

Comment:
Maybe add some info on dimensions here? msg + solution style, since this is fairly user facing